index.js

```diff
@@ -1,8 +1,8 @@
 'use strict';
 
-var processFn = function (fn, P, opts) {
+var processFn = function (fn, P, opts, bind) {
```

Comment:
Yeah. It caused the optimization test to fail (not sure why, but [the optimization killers blog post](https://github.com/petkaantonov/bluebird/wiki/Optimization-killers) does mention something about `fn.bind()`.

Also, we are already calling `fn.apply(someObj, args)` anyways. Adding a layer of indirection where `this` doesn't end up being `someObj` seems confusing.